extensions/sql-database-projects/src/dialogs/deployDatabaseDialog.ts

```diff
@@ -104,14 +104,59 @@ export class DeployDatabaseDialog {
 		});
 	}
 
-	private async deploy(): Promise<void> {
-		// TODO: hook up with build and deploy
+	private async getConnectionUri(): Promise<string> {
 		// if target connection is a data source, have to check if already connected or if connection dialog needs to be opened
+		let connId: string;
+
+		if (this.dataSourcesRadioButton?.checked) {
+			const dataSource = (this.dataSourcesDropDown!.value! as DataSourceDropdownValue).dataSource;
+
+			const connProfile: azdata.IConnectionProfile = {
+				serverName: dataSource.server,
+				databaseName: dataSource.database,
+				connectionName: dataSource.name,
+				userName: '',
+				password: '', // TODO: secure password storage
+				authenticationType: dataSource.integratedSecurity ? 'Integrated' : 'SqlAuth',
+				savePassword: false,
+				providerName: 'MSSQL',
+				saveProfile: true,
+				id: dataSource.name + '-dataSource',
+				options: []
+			};
+
+			const connResult = await azdata.connection.connect(connProfile, false, false);
```

Comment:
does the connection dialog need to pop up if it's SqlAuth?

Reply:
I think for the purposes of this PR (stringing together pieces), it's okay to say that the SqlAuth credentials come from datasources.json, and it the connection will fail if it's missing password.  (see other comment about needing to define more concrete behavior here)

Reply:
So, if datasource connection is picked - and no password is provided - deploy will try to go through and fail? that might feel broken a bit... might be easier to just open the connection dialog and then take a connection from connection dialog. 
userConnection = await azdata.connection.openConnectionDialog(undefined, connectionProfile);


(schema compare scmp open